src/AutoROM.py

```diff
@@ -129,37 +134,61 @@
     "eea0da9b987d661264cce69a7c13c3bd": "zaxxon",
 }
 
-# simply download tar file to specified dir
-def download_tar_to_buffer(url="https://roms8.s3.us-east-2.amazonaws.com/Roms.tar.gz"):
-    with requests.get(url, stream=True) as response:
-        assert response.status_code == 200
-        assert response.headers["Content-Type"] == "application/x-gzip"
-
-        archive_size = int(response.headers["Content-Length"])
-        assert archive_size > 0
-
-        chunk_size = 2 ** 10
-        with tqdm(
-            unit="B",
-            unit_scale=True,
-            unit_divisor=chunk_size,
-            desc="Downloading ROMs",
-            leave=False,
-            total=archive_size,
-        ) as pbar:
-            buffer = io.BytesIO()
-            for chunk in response.iter_content(chunk_size=chunk_size):
-                buffer.write(chunk)
-                pbar.update(len(chunk))
-
-            buffer.flush()
-            buffer.seek(0)
-            return buffer
+
+class ROMDownloadError(Exception):
+    pass
+
+
+# Download ROMs from a torrent magnet link
+def download_tar_to_tmp_file(timeout_seconds=120):
+    state_str = [
+        'queued',
+        'checking',
+        'downloading metadata',
+        'downloading',
+        'finished',
+        'seeding',
+        'unused enum',
+        'checking fastresume',
+    ]
+
+    ses = lt.session()
+    tmp_dir = tempfile.TemporaryDirectory()
```

Comment:
Could you factor this out of `download_tar_to_tmp_file` so you can use the context manager for `tempfile.TemporaryDirectory` so the directory is cleaned up after the files are copied.

This probably involves making `download_tar_to_tmp_file` take an argument `tmp_dir` and then wrapping some things in `main` with the context manager.

Reply:
:+1: done in https://github.com/Farama-Foundation/AutoROM/pull/16/commits/1820bd661e1ef9bdb68381b479b0e71ceb218997